prelude.links

```diff
@@ -456,6 +456,23 @@ fun asList(t) server {
   for (x <-- t) [x]
 }
 
+# Create an arbitrary XML element
+# @param {String} name Name of the element ([a-z]+)
+# @param {[(String, String)]} attributes List of attribute-value-pairs
+# @param {Xml} children Child XML nodes
+sig makeXml : (String, [(String, String)], Xml) ~> Xml
+fun makeXml (name, attributes, children) {
+  var attributes = fold_left(fun (prev, curr) {
+    var (attr, value) = curr;
+    prev ^^ " " ^^ attr ^^ "=\"" ^^ value ^^ "\""
```

Comment:
This has the potential for injection attacks if attributes containing quotes are passed in.  For example one could inject arbitrary XML content inside an element constructed using makeXML by providing an attribute with a value such as 

`xyz\"/> ... arbitrary XML content ...<foo key=\"`

which would give different results from what would happen if we were able to construct XML directly rather than having to go through the parser.

So it should only be used on strings that have been sanitized/checked (parseXML has the same problem, of course...)